service/scheduler.go

```diff
@@ -59,18 +59,38 @@ type Scheduler struct {
 	OrchardHost       string
 	OrchardAPIKeyName string
 	OrchardAPIKey     string
+	LockTTL           time.Duration
 }
 
 func (s *Scheduler) Start() {
 	fmt.Println("Scheduler Started")
 	tick := time.Tick(s.Interval)
-	for range tick {
-		fmt.Println("tick")
-		s.scheduleWorkflows(database.GetInstance())
-		s.activateWorkflows(database.GetInstance())
+	lockTtlTick := time.Tick(time.Minute)
+	for {
+		select {
+		case <-tick:
+			fmt.Println("tick")
+			s.scheduleWorkflows(database.GetInstance())
+			s.activateWorkflows(database.GetInstance())
+		case <-lockTtlTick:
+			fmt.Println("lockTtlTick")
+			s.deleteOldLocks(database.GetInstance())
+		}
 	}
 }
 
+func (s *Scheduler) deleteOldLocks(db *gorm.DB) {
```

Comment:
handles the case where the pod crashes before the lock can be released in the lockAndCreate / lockAndActivate methods